Marker/Classes/Marker.swift

```
@@ -0,0 +1,85 @@
+//
+//  Marker.swift
+//  Marker
+//
+//  Created by Htin Linn on 5/15/17.
+//  Copyright © 2017 Prolific Interactive. All rights reserved.
+//
+
+import Foundation
+
+/// Returns formatted Markdown text as an attributed string.
+///
+/// - Parameters:
+///   - markdownText: String with Markdown tags.
+///   - textStyle: Text style object containing style information.
+/// - Returns: Formatted Markdown text.
+public func attributedMarkdownString(from markdownText: String,
+                                     using textStyle: TextStyle) -> NSAttributedString {
+    guard let (parsedString, elements) = try? MarkdownParser.parse(markdownText) else {
+        return NSAttributedString(string: markdownText, textStyle: textStyle)
```

Is that what we want to return back to the client? Don't we want to propagate an eventual error from the `parse` function?

---

Setting text is a very common operation that you do in your projects. I feel like it would be such as hassle to have to write `try` or `try?` every time you want to use a `Marker` function. What's being returned here is a better compromise to me: the text is still being set, along with the styles being applied to the entire string.

---

I understand the fact that you want to keep the API concise, but at the same time it's misleading to the client using your function. As a client, reading at the function definition, I would not know that if for some reason Marker can't parse my parameters I will have a default behavior.

---

Maybe I should just document that this is the default behavior and define a separate function that returns errors for people who want to handle the error themselves?

---

Yes that works for me!

---

Updated with the changes.